frontend/lib/providers/artefact_builds.dart

```diff
@@ -1,20 +1,40 @@
 import 'package:riverpod_annotation/riverpod_annotation.dart';
 
 import '../models/artefact_build.dart';
+import '../models/test_execution.dart';
 import 'dio.dart';
 
 part 'artefact_builds.g.dart';
 
 @riverpod
-Future<List<ArtefactBuild>> artefactBuilds(
-  ArtefactBuildsRef ref,
-  int artefactId,
-) async {
-  final dio = ref.watch(dioProvider);
+class ArtefactBuilds extends _$ArtefactBuilds {
+  @override
+  Future<List<ArtefactBuild>> build(int artefactId) async {
+    final dio = ref.watch(dioProvider);
 
-  final response = await dio.get('/v1/artefacts/$artefactId/builds');
-  final List artefactBuildsJson = response.data;
-  final artefactBuilds =
-      artefactBuildsJson.map((json) => ArtefactBuild.fromJson(json)).toList();
-  return artefactBuilds;
+    final response = await dio.get('/v1/artefacts/$artefactId/builds');
+    final List artefactBuildsJson = response.data;
+    final artefactBuilds =
+        artefactBuildsJson.map((json) => ArtefactBuild.fromJson(json)).toList();
+    return artefactBuilds;
+  }
+
+  Future<void> changeReviewDecision(
+    int testExecutionId,
+    String reviewComment,
+    List<TestExecutionReviewDecision> reviewDecision,
+  ) async {
+    final dio = ref.watch(dioProvider);
+
+    await dio.patch(
+      '/v1/test-executions/$testExecutionId',
+      data: {
+        'review_decision': reviewDecision.map((e) => e.toJson()).toList(),
+        'review_comment': reviewComment,
+      },
+    );
+
+    final updatedState = await build(artefactId);
+    state = AsyncData(updatedState);
```

Comment:
I was also thinking about this, but the issue with the first option was that the PATCH endpoint modified the TestExecution object, not the ArtefactBuild object, and the riverpod provider is for the ArtefactBuilds. 

I guess one option would be to have this endpoint return the whole ArtefactBuild object, which is a bit counter-intuitive as we are sending PATCH request to a TestExecution object.

Reply:
True, I hadn't taken that into consideration. Then I guess options 2 or 3 would work. Where for 3 we would just replace the child test execution object in the artefact build

Reply:
I was thinking it would be cleaner to go with the second approach, although it involves an extra network request to rebuild the state, but it makes sure we always put the correct data in the state.